docs/reference/modules/node.asciidoc

```diff
@@ -1,32 +1,304 @@
 [[modules-node]]
 == Node
 
-*elasticsearch* allows to configure a node to either be allowed to store
-data locally or not. Storing data locally basically means that shards of
-different indices are allowed to be allocated on that node. By default,
-each node is considered to be a data node, and it can be turned off by
-setting `node.data` to `false`.
-
-This is a powerful setting allowing to simply create smart load
-balancers that take part in some of different API processing. Lets take
-an example:
-
-We can start a whole cluster of data nodes which do not even start an
-HTTP transport by setting `http.enabled` to `false`. Such nodes will
-communicate with one another using the
-<<modules-transport,transport>> module. In front
-of the cluster we can start one or more "non data" nodes which will
-start with HTTP enabled. All HTTP communication will be performed
-through these "non data" nodes.
-
-The benefit of using that is first the ability to create smart load
-balancers. These "non data" nodes are still part of the cluster, and
-they redirect operations exactly to the node that holds the relevant
-data. The other benefit is the fact that for scatter / gather based
-operations (such as search), these nodes will take part of the
-processing since they will start the scatter process, and perform the
-actual gather processing.
-
-This relieves the data nodes to do the heavy duty of indexing and
-searching, without needing to process HTTP requests (parsing), overload
-the network, or perform the gather processing.
+Any time that you start an instance of Elasticsearch, you are starting a
+**node** of Elasticsearch. A collection of connected nodes is what is
+called a <<modules-cluster,cluster>> of Elasticsearch. If you are running a
+single node of Elasticsearch, then you have a cluster of one node.
+
+Every node in the cluster can handle <<modules-http,HTTP>> and
+<<modules-transport,Transport>> traffic by default. The transport layer
+is used exclusively for communication from node-to-node and the
+{javaclient}/transport-client.html[Java `TransportClient`]; the HTTP layer
+is used only from external requests.
+
+Each node in the cluster serves one or more purposes, and there are
+four types of nodes:
+
+[cols="<,<,<,<",options="header",]
+|=======================================================================
+|Node Type |Setting |Default |Description
+|<<master-node,Master>> |`node.master` |`true` |A node that is eligible
+ to control the cluster.
+|<<data-node,Data>> |`node.data` |`true` |A node that stores index data
+ for search activity.
+|<<client-node,Client>> |(All Above) |N/A |A standalone node that routes
+requests, but does not hold data or handle master duties.
+
+All nodes offer this behavior at a minimum, but they perform other tasks (e.g.,
+`node.data: true`) that must be disabled to be a "true" client node.
```

Comment:
Agreed. I struggled to come up with wording for awhile to avoid using `node.client`. This was the fruits of that labor.